File: app/src/main/java/com/delivery/deliveryapp/models/Menu.java
package com.delivery.deliveryapp.models;

import java.io.Serializable;
import java.util.ArrayList;

public class Menu implements Serializable{

    private String name;
    private ArrayList<Dish> dishes;

    public Menu(String name)
    {
        this.name = name;
        dishes = new ArrayList<Dish>();
    }

    public Menu(String name, ArrayList<Dish> dishes)
    {
        this.name = name;
        this.dishes = dishes;
    }

    public String getName() {return this.name;}
    public ArrayList<Dish> getDishes() {return this.dishes;}

    public void add(Dish dish)
    {
        this.dishes.add(dish);
    }
}

File: app/src/main/java/com/delivery/deliveryapp/models/Menu.kt
package com.delivery.deliveryapp.models

import java.io.Serializable
import java.util.ArrayList

class Menu(val name: String) : Serializable {

    var dishes: ArrayList<Dish>
        private set


    init {
        this.dishes = ArrayList()
    }

    constructor(name: String, dishes: ArrayList<Dish>) : this(name) {
        this.dishes = dishes
    }

    fun add(dish: Dish) {
        this.dishes.add(dish)
    }
}
